sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -0,0 +1,1218 @@
+  {
+    exit(1);
+  }
+
+  *out_is_max_temp_changed = false;
+  device_current_temperature = temperature;
+
+  // Update maximum or minimum temperatures.
+  if (device_current_temperature > device_maximum_temperature)
+  {
+    device_maximum_temperature = device_current_temperature;
+    *out_is_max_temp_changed = true;
+  }
+  else if (device_current_temperature < device_minimum_temperature)
+  {
+    device_minimum_temperature = device_current_temperature;
+  }
+
+  // Calculate the new average temperature.
+  device_temperature_count++;
+  device_temperature_summation += device_current_temperature;
+  device_average_temperature = device_temperature_summation / device_temperature_count;
+
+  IOT_SAMPLE_LOG_SUCCESS("Client updated desired temperature variables locally.");
+  IOT_SAMPLE_LOG("Current Temperature: %2f", device_current_temperature);
+  IOT_SAMPLE_LOG("Maximum Temperature: %2f", device_maximum_temperature);
+  IOT_SAMPLE_LOG("Minimum Temperature: %2f", device_minimum_temperature);
+  IOT_SAMPLE_LOG("Average Temperature: %2f", device_average_temperature);
+}
+
+static void send_reported_property(az_span name, double value, int32_t version, bool confirm)
+{
+  az_result rc;
+
+  // Get the property PATCH topic to send a reported property update.
+  char property_patch_topic_buffer[128];
+  rc = az_iot_pnp_client_property_patch_get_publish_topic(
+      &pnp_client,
+      get_request_id(),
+      property_patch_topic_buffer,
+      sizeof(property_patch_topic_buffer),
+      NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to get the property PATCH topic");
+
+  // Build the updated reported property message.
+  char reported_property_payload_buffer[128];
+  az_span reported_property_payload = AZ_SPAN_FROM_BUFFER(reported_property_payload_buffer);
+
+  if (confirm)
+  {
+    build_property_payload_with_status(
+        name,
+        value,
+        AZ_IOT_STATUS_OK,
+        version,
+        property_success_name,
+        reported_property_payload,
+        &reported_property_payload);
+  }
+  else
+  {
+    uint8_t count = 1;
+    az_span const names[1] = { name };
+    double const values[1] = { value };
+
+    build_property_payload(
+        count, names, values, NULL, reported_property_payload, &reported_property_payload);
+  }
+
+  // Publish the reported property update.
+  publish_mqtt_message(
+      property_patch_topic_buffer, reported_property_payload, IOT_SAMPLE_MQTT_PUBLISH_QOS);
+  IOT_SAMPLE_LOG_SUCCESS("Client published the property PATCH reported property message.");
+  IOT_SAMPLE_LOG_AZ_SPAN("Payload:", reported_property_payload);
+}
+
+static void handle_command_request(
+    MQTTClient_message const* message,
+    az_iot_pnp_client_command_request const* command_request)
+{
+  az_span const message_span = az_span_create((uint8_t*)message->payload, message->payloadlen);
+
+  if (az_span_is_content_equal(command_getMaxMinReport_name, command_request->command_name))
```

Comment:
Whenever I see calls to `az_span_is_content_equal` when we have JSON strings being processed, I have to do a double take. 

Are the parameters received from a JSON payload being parsed? If so, consider adding a comment here, why byte-by-byte comparison is what we want to do here.

Reply:
The commands are not JSON payloads. These are pulled from an MQTT topic so should be byte by byte